api/slackimport.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api
+
+import (
+	"archive/zip"
+	"bytes"
+	l4g "code.google.com/p/log4go"
+	"encoding/json"
+	"github.com/mattermost/platform/model"
+	"io"
+	"mime/multipart"
+	"strconv"
+	"strings"
+)
+
+type SlackChannel struct {
+	Id      string            `json:"id"`
+	Name    string            `json:"name"`
+	Members []string          `json:"members"`
+	Topic   map[string]string `json:"topic"`
+}
+
+type SlackUser struct {
+	Id       string            `json:"id"`
+	UserName string            `json:"name"`
+	Profile  map[string]string `json:"profile"`
+}
+
+type SlackPost struct {
+	User        string            `json:"user"`
+	BotId       string            `json:"bot_id"`
+	BotUsername string            `json:"username"`
+	Text        string            `json:"text"`
+	TimeStamp   string            `json:"ts"`
+	Type        string            `json:"type"`
+	SubType     string            `json:"subtype"`
+	Comment     map[string]string `json:"comment"`
+}
+
+func SlackConvertTimeStamp(ts string) int64 {
+	timeString := strings.SplitN(ts, ".", 2)[0]
+
+	timeStamp, err := strconv.ParseInt(timeString, 10, 64)
+	if err != nil {
+		l4g.Warn("Bad timestamp detected")
+		return 1
+	}
+	return timeStamp * 1000 // Convert to milliseconds
+}
+
+func SlackParseChannels(data io.Reader) []SlackChannel {
+	decoder := json.NewDecoder(data)
+
+	var channels []SlackChannel
+	if err := decoder.Decode(&channels); err != nil {
+		return make([]SlackChannel, 0)
+	}
+	return channels
+}
+
+func SlackParseUsers(data io.Reader) []SlackUser {
+	decoder := json.NewDecoder(data)
+
+	var users []SlackUser
+	if err := decoder.Decode(&users); err != nil {
+		return make([]SlackUser, 0)
+	}
+	return users
+}
+
+func SlackParsePosts(data io.Reader) []SlackPost {
+	decoder := json.NewDecoder(data)
+
+	var posts []SlackPost
+	if err := decoder.Decode(&posts); err != nil {
+		return make([]SlackPost, 0)
+	}
+	return posts
+}
+
+func SlackAddUsers(teamId string, slackusers []SlackUser, log *bytes.Buffer) map[string]*model.User {
+	// Log header
+	log.WriteString("\n Users Created\n")
+	log.WriteString("===============\n\n")
+
+	// Add users
```

Comment:
Not sure if this is a particularly useful comment since the function name is SlackAddUsers :P